bookstore/archive.py

```diff
@@ -52,12 +107,16 @@ def run_pre_save_hook(self, model, path, **kwargs):
         if model["type"] != "notebook":
             return
 
-        # TODO: store the hash of the notebook to not write on every save
-        notebook_contents = json.dumps(model["content"])
+        content = json.dumps(model["content"])
 
-        full_path = self.s3_path(path)
+        # run_pre_save_hook isn't async, though this should be awaited. I need to look into how to integrate with
```

Comment:
Sill noodling on this. When we are trying to put things into the queue to tee up a save we can call [`put_nowait()`](https://www.tornadoweb.org/en/stable/queues.html#tornado.queues.Queue.put_nowait) which will just tell us no, which is ok because we are too busy to try saving and we are only attempting to archive. The auto save from Jupyter will try again later if the user has the notebook open, right (is that always enabled)? If that's not always true to eventually clean up when the queue is full then I think we will need something like their `fetching` set and a little fancier arrangement.

Reply:
Yeah the `put_nowait` makes sense in this case, then we'd just catch the `QueueFull` exception. We'd still let this function terminate so the regular save happens.